src/refract/RenderJSONVisitor.cc

```diff
@@ -77,6 +79,56 @@ namespace refract
         }
     }
 
+    template<typename T>
+    const T* getDefault(const T& e) {
+        IElement::MemberElementCollection::const_iterator i = e.attributes.find("default");
+        if (i == e.attributes.end()) {
+            return NULL;
+        }
+        TypeQueryVisitor v;
+        return TypeQueryVisitor::as<T>((*i)->value.second);
+    }
+
+    template<typename T>
+    const T* getSample(const T& e) {
+        IElement::MemberElementCollection::const_iterator i = e.attributes.find("samples");
+        if (i == e.attributes.end()) {
+            return NULL;
+        }
+
+        TypeQueryVisitor v;
+        ArrayElement* a = TypeQueryVisitor::as<ArrayElement>((*i)->value.second);
+
+        if (!a || a->value.empty()) {
+            return NULL;
+        }
+
+        return TypeQueryVisitor::as<T>(*(a->value.begin()));
+
+    }
+
+    template<typename T, typename R = typename T::ValueType>
+    struct getValue {
+        const T& element;
+
+        getValue(const T& e) : element(e) {}
+
+        operator const R*() {
+
+            if (const T* d = getDefault(element)) {
```

Comment:
@klokane Why are we giving more importance to `default` values? I think sample values should get more importance in rendering since .... well they are samples.